lib/collection/src/collection/shard_transfer.rs

```diff
@@ -250,10 +252,14 @@ impl Collection {
                 )));
             };
 
+            // Wait for the replica set to have the local shard initialized
+            // This can take some time as this is arranged through consensus
+            replica_set.wait_for_local(CONSENSUS_DEFAULT_META_OP_WAIT);
```

Comment:
This is a blocking call (it comes down to waiting on a `parking_lot::RwLock`) in async context. :/

Reply:
(But, I think, it should be easy to wrap this with `tokio::task::spawn_blocking`.)

Reply:
Good catch. Implemented with `spawn_blocking` within `ShardReplicaSet::wait_for` in abd85b863c959a08a5f1de42ca11ef474536d3bb.